.github/workflows/wheels.yml

```diff
@@ -153,11 +153,14 @@ jobs:
         with:
           python-version: '3.9'  # update once build dependencies are available
 
-      - name: Set pip extra index
+      - name: Set pip extra options
         if: github.event_name == 'schedule'
         env:
           PIP_EXTRA_INDEX: https://pypi.anaconda.org/scipy-wheels-nightly/simple
-        run: echo ${{ env.PIP_EXTRA_INDEX }}
+          PIP_PRE: 1
+        run: |
+          echo "PIP_EXTRA_INDEX=${{ env.PIP_EXTRA_INDEX }}" >> $GITHUB_ENV
+          echo "PIP_PRE=${{ env.PIP_PRE }}" >> $GITHUB_ENV
```

Comment:
Nit: Since this only runs when `github.event_name == 'schedule'`, we can set the variables directly and not go through the indirection of `env`:

```suggestion
          # Test with nightly dependencies for nightly builds
          echo "PIP_EXTRA_INDEX=https://pypi.anaconda.org/scipy-wheels-nightly/simple" >> $GITHUB_ENV
          echo "PIP_PRE=1" >> $GITHUB_ENV
```

Reply:
Ok, thanks, will do.
Let me just check if it works that way with respect to your [comment](https://github.com/scikit-learn/scikit-learn/pull/24446#discussion_r992818171) (I'm testing on a personal fork).
I'm wondering if putting the variable one level up with an if condition will apply the condition to the entire job... Am I correct?

Reply:
It worked [here](https://github.com/cmarmo/scikit-learn/actions/runs/3232118935).